### PR TITLE
Add Metric: Beanstalk Deploy

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description

Adding a metric for deploying to Beanstalk, to be used in the VS Toolkit

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -119,14 +119,39 @@
...
         {
-          "name": "regionId",
-          "type": "string",
-          "description": "The ID of the region that was selected"
+            "name": "regionId",
+            "type": "string",
+            "description": "The ID of the region that was selected"
+        },
+        {
+            "name": "partitionId",
```
Comment (hunterwerlla) [inline telemetry/definitions/commonDefinitions.json:127]: not used? also not needed

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -119,14 +119,39 @@
...
+            "type": "string",
+            "description": "The ID of the partition that was selected"
+        },
+        {
+            "name": "redeploy",
```
Comment (hunterwerlla) [inline telemetry/definitions/commonDefinitions.json:132]: Can we do the opposite of redeploy, "initial" or something along those lines?

File: telemetry/definitions/commonDefinitions.json [hunk 3]
```diff
@@ -119,14 +119,39 @@
...
+            "type": "boolean",
+            "description": "Whether or not the deploy targets an existing destination (true) or a new destination (false)"
+        },
+        {
+            "name": "solutionStack",
```
Comment (hunterwerlla) [inline telemetry/definitions/commonDefinitions.json:137]: Can we use "name"?

File: telemetry/definitions/commonDefinitions.json (hunk 1 above)
Comment (awschristou) [reply telemetry/definitions/commonDefinitions.json:127]: partitionId was already present, but didn't have the indentation formatting

File: telemetry/definitions/commonDefinitions.json (hunk 3 above)
Comment (awschristou) [reply telemetry/definitions/commonDefinitions.json:137]: Sure

File: telemetry/definitions/commonDefinitions.json (hunk 1 above)
Comment (hunterwerlla) [reply telemetry/definitions/commonDefinitions.json:127]: Ah I see it's used in setPartition, that's new so didn't know it was there

**Pull request merged**